contents/security/sqlInjection.md

```diff
@@ -0,0 +1,181 @@
+# SQL Injection #
+
+Author: [Lewis Koh](https://github.com/nus-cs3281/2018/blob/master/students/lewisKoh/lewisKoh-Resume.md)
+
+## Overview ##
+
+SQL (Structured Query Language) Injection is a vulnerability found in websites using databases.
+
+SQL is a common language which is used by websites to communicate with databases. Databases 
+can be used to store persistent data, such as usernames and passwords, sensitive account data, 
+or other important information used by the website. Databases are usually made of many "tables" 
+which are organised in rows and columns. Each row is a separate entry in the table, and 
+each column is a specific parameter which can be used by the entry.
+
+As an example, a table in a database may look like this:
+
+
+<table>
+	<tr>
+		<th>UserId</th>
+		<th>Username</th>
+		<th>Password</th>
+	</tr>
+	<tr>
+		<td>1</td>
+		<td>Admin</td>
+		<td>123456</td>
+	</tr>
+	<tr>
+		<td>2</td>
+		<td>Alice</td>
+		<td>password</td>
+	</tr>
+	<tr>
+		<td>3</td>
+		<td>...</td>
+		<td>...</td>
+	</tr>
+</table>
+
+
+SQL is used to interact with the database by sending "queries" which the database responds to. 
+Some common SQL commands used include: 
+* SELECT - retrieves information from a table
+* UPDATE - changes information from a table
+* INSERT INTO - adds a new entry into a table
+* DELETE FROM - deletes information from a table
+
+Websites combine SQL syntax with user input to query the underlying database of the website. 
+For example, blog websites may ask for a username and password from users for them to log 
+in to the site. This input is used to create the SQL statement that will be sent to the database. 
+A website which is vulnerable to SQL injection attacks may construct the SQL query like this:
+
+```sql
+String query = "SELECT * FROM Users WHERE Username = ‘" + input_username +"’ 
+                AND Password = ‘" + input_password + "’";
+```
+
+As an example, if a user entered their username, "Alice", and password, "password" into the website 
+to try to gain access, the constructured query would look like this:
+
+```sql
+SELECT * FROM Users WHERE Username = ‘Alice’ 
+AND Password = ‘password’
+```
+
+This query would look at the "Users" table in the database, and return any entries where the 
+column "Username" matches "Alice" and the column "Password" matches "password". If the previous 
+table was searched, it would return the second row as the result. The website would be able to tell 
+that a legitimate username and password was entered, and would let the user log into the site.
+
+
+However, are users limited to just filling in legitimate username and password strings? Some websites 
+directly use the input given by the user without any modifications, thus the users are able to 
```

Comment:
Perhaps you could say.. users are able to write code as input To make it flow better.

Also i think a more accurate phrasing is validation/sanitation instead of modification which suggests that the valid input is also changed.